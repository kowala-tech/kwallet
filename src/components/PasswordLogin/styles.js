const styles = (theme) => ({
	flexContainer: {
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "center",
		textAlign: "center"
	},
	errorMessage: {
		paddingBottom: theme.spacing.unit * 6,
		height: theme.spacing.unit * 4
	},
	text: {
		fontWeight: 300
	},
});

export default styles;
