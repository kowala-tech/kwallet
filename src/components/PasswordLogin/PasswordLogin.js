import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
// Component Related Imports
import styles from "./styles";
import PasswordForm from "./components/PasswordForm";

class PasswordLogin extends React.Component {
	render() {
		const {
			classes,
			handleSubmit,
			loading,
			errorMessage
		} = this.props;

		return (
			<div className={classes.flexContainer}>
				<Typography className={classes.text}>{errorMessage}</Typography>
				<PasswordForm
					loading={loading}
					onSubmit={handleSubmit}
				/>
			</div>
		);
	}

}

PasswordLogin.propTypes = {
	classes: PropTypes.object.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.bool.isRequired,
	errorMessage: PropTypes.string
};

export default compose(
	withStyles(styles)
)(PasswordLogin);
