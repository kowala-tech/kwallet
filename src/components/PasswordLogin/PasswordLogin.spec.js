import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import PasswordLogin from "./PasswordLogin";

describe("PasswordLogin", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<PasswordLogin
					handleSubmit={jest.fn()}
					loading={false}
					error={false}
					errorMessage={""}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
