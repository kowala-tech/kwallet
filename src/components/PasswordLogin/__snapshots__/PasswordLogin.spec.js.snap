// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordLogin it should match the snapshot 1`] = `
<PasswordLogin
  classes={
    Object {
      "errorMessage": "PasswordLogin-errorMessage-2",
      "flexContainer": "PasswordLogin-flexContainer-1",
      "text": "PasswordLogin-text-3",
    }
  }
  error={false}
  errorMessage=""
  handleSubmit={[MockFunction]}
  loading={false}
/>
`;
